src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -1200,6 +1202,29 @@ private boolean confirmClose(LibraryTab libraryTab) {
             // Save was cancelled or an error occurred.
             return false;
         }
+        if (response.isPresent() && response.get().equals(discardChanges)) {
+            //filename = filename + "--discarded.bak";
+
+            Path directory = Path.of(filename);
+            try {
+                SaveDatabaseAction saveAction = new SaveDatabaseAction(libraryTab, prefs, Globals.entryTypesManager);
```

Comment:
Do NOT Call UI methods for saving. The logic for creating a `--discared` file should be in the BackupManager. You can see the saving at org.jabref.logic.autosaveandbackup.BackupManager#performBackup.

As written in https://github.com/JabRef/jabref/issues/9361#issuecomment-1343957665, it really should be done in BackupManager.